### PR TITLE
1.x Frontend Work

### LINKED ISSUE #328: Add Images to Home Page
I know Ben had some solid graphics he created for a mock up, or maybe in a branch for a later release, but I'd like to just roll those out now.

### LINKED ISSUE #329: Clean Up Navbar
The way we have things right now is **super** ugly. I'd prefer for it to be:

- Explore
  - Data Sets
  - Array of Things
  - View a list of Data Sets (requires a new controller/template)
- Developers
  - Documentation
  - API
  - GitHub
- {user or sign in}
  - Home
  - Add a New Data Set
  - (divider)
  - Sign Out
- {admin}

I'd like @sanilio and @HeyZoos to weigh in on this.

### LINKED ISSUE #379: Tooltip Icons not Rendering
Getting a garbled icon/character for the tooltip indicator.

## Are there any specific error messages?
No -- _console_ and _network_ come up clean.

## Browsers
Firefox and Chromium on Linux

screenshot_2018-07-23 plenario 1

### PR DESCRIPTION
**Pull request by brlodi**
This closes #328 and closes #329, including quite a few structural changes to accommodate it (they needed to be made anyway, and I didn't want to do it "wrong" first only to fix it later).

Should also resolve #379 as a side effect.

I didn't implement automatic deployment of generated images to S3, but that's something I could do if requested. As it is I don't anticipate regenerating the images often, only if we want to swap one out, so it may not be worth it. Up to @vforgione.

Sorry again that this took so long. It involved a few exploratory things that weren't as directly suited to implementing as they might have seemed, but it should at least mesh with our future frontend plans so that effort wasn't entirely extraneous.

### CODE REVIEW

**Comment by brlodi** [comment]
I can redo this with a squashed commit if that would be preferred.

**Comment by vforgione** [comment]
Nah. That's fine. I don't really mind messy histories.

**Comment by brlodi** [comment]
Ugh, this is going to have to wait for me to figure something out.

Phoenix's handy-dandy helper to create POST links (by automatically generating and submitting a form via JavaScript) don't work if anything else assigns JavaScript to them, because it overrides the handler one way or another. Like, say, most of Bootstrap. I'm going to have to write some code to work around that, probably manually reimplementing the majority of Phoenix's stock JavaScript.

(I don't think anyone actually tests Phoenix in real-world applications. The number of these obnoxious conflicts is too high.)

**tl;dr** because `logout` is a POST action we just literally can't put it in a dropdown, since Phoenix has JS side effects that they apparently see no reason to document because it's a secondary, inferior language.

**Comment by vforgione** [comment]
screenshot_2018-07-24 plenario

**Comment by brlodi** [comment]
Sorry about the link, I missed it when I updated the others.

---

As for the Apiary rendering issue, I'm actually surprised we didn't have the problem before. Apiary uses an iframe with the attribute `height='100%'`, which _does not work_ in HTML5. As far as I can tell, nothing in this PR should change how it behaves. Also, I don't think any amount of effort will make Apiary look *good* embedded in our page. It will either bleed all the way to the edges of the screen and look dumb with our navbar, or exist inside our container structure and look dumb because Apiary assumes it takes over the whole page.

> By default, embed takes over the whole browser viewport.

image


That looks kind of dumb, but putting it in a container does this:

image

and also necessarily either makes the whole page scrollable or forces ugly gaps between the navbar and docs and docs and bottom of page. Fixing them would require rewriting most of the HTML and SCSS for the site.

@HeyZoos I'm going to make Apiary styling your responsibility, because it's your baby and I don't want to have to mess with trying to make it behave. Nevertheless, I've added several lines of crap to our SCSS to somewhat solve the problem, at least temporarily.

**Comment by vforgione** [comment]
Regarding Apiary, why are we embedding it anyway? I didn't really like it to begin with, but I wasn't about to make that an issue. It's now become a problem.

Why not just make the nav item link to the _actual_ docs on Apiary with a `target="_blank"`? It seems silly to embed it.

**Comment by HeyZoos** [comment]
If it's more trouble than it's worth let's just do a link then.

**Comment by brlodi** [comment]
@vforgione please give this another look when you get a chance. Just needs the version number bump before we merge it.
